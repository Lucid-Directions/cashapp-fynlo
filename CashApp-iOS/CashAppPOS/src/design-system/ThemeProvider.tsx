import React, { createContext, useContext, useState, useEffect, ReactNode } from 'react';
import { Appearance, ColorSchemeName } from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { Theme, lightTheme, darkThemeConfig } from './theme';

// Theme mode types
export type ThemeMode = 'light' | 'dark' | 'auto';
export type ColorTheme = 'default' | 'blue' | 'purple' | 'orange' | 'red' | 'teal' | 'indigo' | 'pink' | 'lime' | 'amber';

// Color theme options interface
export interface ColorThemeOption {
  id: ColorTheme;
  label: string;
  primary: string;
  secondary: string;
  accent: string;
  description: string;
}

// Theme context interface
interface ThemeContextType {
  theme: Theme;
  themeMode: ThemeMode;
  colorTheme: ColorTheme;
  isDark: boolean;
  setThemeMode: (mode: ThemeMode) => void;
  setColorTheme: (colorTheme: ColorTheme) => void;
  toggleTheme: () => void;
}

// Create context
const ThemeContext = createContext<ThemeContextType | undefined>(undefined);

// Storage keys for theme preferences
const THEME_STORAGE_KEY = 'fynlo_theme_mode';
const COLOR_THEME_STORAGE_KEY = 'fynlo_color_theme';

// Color theme definitions
const colorThemeOptions: ColorThemeOption[] = [
  {
    id: 'default',
    label: 'Fynlo Green',
    primary: '#00A651',
    secondary: '#0066CC',
    accent: '#22C55E',
    description: 'Classic Fynlo brand colors',
  },
  {
    id: 'blue',
    label: 'Ocean Blue',
    primary: '#0EA5E9',
    secondary: '#1E40AF',
    accent: '#3B82F6',
    description: 'Calming ocean blue theme',
  },
  {
    id: 'purple',
    label: 'Royal Purple',
    primary: '#8B5CF6',
    secondary: '#7C3AED',
    accent: '#A855F7',
    description: 'Elegant purple theme',
  },
  {
    id: 'orange',
    label: 'Fynlo Orange',
    primary: '#FF6D00',
    secondary: '#121212',
    accent: '#FF8F00',
    description: 'Official Fynlo brand colours',
  },
  {
    id: 'red',
    label: 'Cherry Red',
    primary: '#EF4444',
    secondary: '#DC2626',
    accent: '#F87171',
    description: 'Bold cherry red theme',
  },
  {
    id: 'teal',
    label: 'Emerald Teal',
    primary: '#14B8A6',
    secondary: '#0F766E',
    accent: '#2DD4BF',
    description: 'Fresh emerald teal',
  },
  {
    id: 'indigo',
    label: 'Deep Indigo',
    primary: '#6366F1',
    secondary: '#4F46E5',
    accent: '#818CF8',
    description: 'Deep indigo blue',
  },
  {
    id: 'pink',
    label: 'Rose Pink',
    primary: '#EC4899',
    secondary: '#DB2777',
    accent: '#F472B6',
    description: 'Elegant rose pink',
  },
  {
    id: 'lime',
    label: 'Fresh Lime',
    primary: '#84CC16',
    secondary: '#65A30D',
    accent: '#A3E635',
    description: 'Fresh lime green',
  },
  {
    id: 'amber',
    label: 'Golden Amber',
    primary: '#F59E0B',
    secondary: '#D97706',
    accent: '#FBBF24',
    description: 'Warm golden amber',
  },
];

// Theme provider props
interface ThemeProviderProps {
  children: ReactNode;
  defaultTheme?: ThemeMode;
  defaultColorTheme?: ColorTheme;
}

export const ThemeProvider: React.FC<ThemeProviderProps> = ({
  children,
  defaultTheme = 'light',
  defaultColorTheme = 'default'
}) => {
  const [themeMode, setThemeModeState] = useState<ThemeMode>(defaultTheme);
  const [colorTheme, setColorThemeState] = useState<ColorTheme>(defaultColorTheme);
  const [systemColorScheme, setSystemColorScheme] = useState<ColorSchemeName>(
    Appearance.getColorScheme()
  );

  // Apply color theme to base theme
  const applyColorTheme = (baseTheme: Theme, colorThemeId: ColorTheme): Theme => {
    const colorOption = colorThemeOptions.find(option => option.id === colorThemeId);
    if (!colorOption) return baseTheme;

    return {
      ...baseTheme,
      colors: {
        ...baseTheme.colors,
        primary: colorOption.primary,
        secondary: colorOption.secondary,
        accent: colorOption.accent,
      }
    };
  };

  // Calculate current theme based on mode, system preference, and color theme
  const calculateCurrentTheme = (mode: ThemeMode, systemScheme: ColorSchemeName, colorThemeId: ColorTheme): Theme => {
    let baseTheme: Theme;
    if (mode === 'auto') {
      baseTheme = systemScheme === 'dark' ? darkThemeConfig : lightTheme;
    } else {
      baseTheme = mode === 'dark' ? darkThemeConfig : lightTheme;
    }
    
    return applyColorTheme(baseTheme, colorThemeId);
  };

  const [currentTheme, setCurrentTheme] = useState<Theme>(
    calculateCurrentTheme(themeMode, systemColorScheme, colorTheme)
  );

  // Load theme preferences from storage
  useEffect(() => {
    const loadThemePreferences = async () => {
      try {
        const [savedTheme, savedColorTheme] = await Promise.all([
          AsyncStorage.getItem(THEME_STORAGE_KEY),
          AsyncStorage.getItem(COLOR_THEME_STORAGE_KEY)
        ]);
        
        if (savedTheme && ['light', 'dark', 'auto'].includes(savedTheme)) {
          setThemeModeState(savedTheme as ThemeMode);
        }
        
        if (savedColorTheme && colorThemeOptions.find(option => option.id === savedColorTheme)) {
          // If orange theme is stored, reset to default green theme
          if (savedColorTheme === 'orange') {
            setColorThemeState('default');
            await AsyncStorage.setItem(COLOR_THEME_STORAGE_KEY, 'default');
          } else {
            setColorThemeState(savedColorTheme as ColorTheme);
          }
        }
      } catch (error) {
        console.warn('Failed to load theme preferences:', error);
      }
    };

    loadThemePreferences();
  }, []);

  // Listen to system color scheme changes
  useEffect(() => {
    const subscription = Appearance.addChangeListener(({ colorScheme }) => {
      setSystemColorScheme(colorScheme);
    });

    return () => subscription?.remove();
  }, []);

  // Update current theme when mode, color theme, or system scheme changes
  useEffect(() => {
    const newTheme = calculateCurrentTheme(themeMode, systemColorScheme, colorTheme);
    setCurrentTheme(newTheme);
  }, [themeMode, systemColorScheme, colorTheme]);

  // Set theme mode and persist to storage
  const setThemeMode = async (mode: ThemeMode) => {
    try {
      setThemeModeState(mode);
      await AsyncStorage.setItem(THEME_STORAGE_KEY, mode);
    } catch (error) {
      console.warn('Failed to save theme preference:', error);
    }
  };

  // Set color theme and persist to storage
  const setColorTheme = async (colorThemeId: ColorTheme) => {
    try {
      setColorThemeState(colorThemeId);
      await AsyncStorage.setItem(COLOR_THEME_STORAGE_KEY, colorThemeId);
    } catch (error) {
      console.warn('Failed to save color theme preference:', error);
    }
  };

  // Toggle between light and dark mode
  const toggleTheme = () => {
    const newMode = currentTheme.isDark ? 'light' : 'dark';
    setThemeMode(newMode);
  };

  const contextValue: ThemeContextType = {
    theme: currentTheme,
    themeMode,
    colorTheme,
    isDark: currentTheme.isDark,
    setThemeMode,
    setColorTheme,
    toggleTheme,
  };

  return (
    <ThemeContext.Provider value={contextValue}>
      {children}
    </ThemeContext.Provider>
  );
};

// Hook to use theme context
export const useTheme = (): ThemeContextType => {
  const context = useContext(ThemeContext);
  if (context === undefined) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  return context;
};

// HOC for components that need theme
export function withTheme<P extends object>(
  Component: React.ComponentType<P & { theme: Theme }>
) {
  return function ThemedComponent(props: P) {
    const { theme } = useTheme();
    return <Component {...props} theme={theme} />;
  };
}

// Utility hook for creating themed styles
export function useThemedStyles<T>(
  createStyles: (theme: Theme) => T
): T {
  const { theme } = useTheme();
  return React.useMemo(() => createStyles(theme), [theme, createStyles]);
}

// Style factory helper
export const createThemedStyles = <T>(
  styleFactory: (theme: Theme) => T
) => {
  return (theme: Theme): T => styleFactory(theme);
};

// Export color theme options for use in components
export { colorThemeOptions };

export default ThemeProvider;