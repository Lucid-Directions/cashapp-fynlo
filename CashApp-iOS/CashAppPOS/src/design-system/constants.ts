// Theme Constants for Fynlo POS Design System
// Centralized constants to avoid magic numbers in components

export const ICON_SIZES = {
  xs: 16,
  sm: 20,
  md: 24,
  lg: 32,
  xl: 48,
  xxl: 60,
  xxxl: 80,
} as const;

export const MODAL_DIMENSIONS = {
  small: {
    width: 320,
    maxHeight: 400,
  },
  medium: {
    width: 400,
    maxHeight: 500,
  },
  large: {
    width: 500,
    maxHeight: 600,
  },
  fullscreen: {
    widthPercent: 0.9,
    maxWidth: 500,
    heightPercent: 0.8,
  },
} as const;

export const ANIMATION_DURATIONS = {
  instant: 0,
  fast: 200,
  normal: 300,
  slow: 500,
  verySlow: 1000,
} as const;

export const ANIMATION_CONFIG = {
  spring: {
    friction: 8,
    tension: 40,
  },
  timing: {
    useNativeDriver: true,
  },
} as const;

export const Z_INDEX = {
  base: 0,
  dropdown: 100,
  sticky: 200,
  overlay: 300,
  modal: 400,
  popover: 500,
  tooltip: 600,
  toast: 700,
} as const;

export const OPACITY = {
  disabled: 0.5,
  overlay: 0.7,
  subtle: 0.8,
  full: 1,
} as const;

export const BORDER_WIDTH = {
  thin: 1,
  medium: 2,
  thick: 3,
} as const;

export const LINE_HEIGHT = {
  tight: 18,
  normal: 22,
  relaxed: 26,
  loose: 30,
} as const;

export const FONT_SIZES = {
  xs: 12,
  sm: 14,
  md: 16,
  lg: 18,
  xl: 20,
  xxl: 24,
  xxxl: 36,
} as const;

export const PROGRESS_BAR = {
  height: 4,
  borderRadius: 2,
} as const;

export const LETTER_SPACING = {
  tight: -1,
  normal: 0,
  wide: 1,
} as const;
