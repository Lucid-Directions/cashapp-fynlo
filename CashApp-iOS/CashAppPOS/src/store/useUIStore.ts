import { create } from 'zustand';

import type { UIState } from '../types';

interface UIStore extends UIState {
  // Category actions
  setSelectedCategory: (category: string) => void;

  // Modal actions
  setShowPaymentModal: (show: boolean) => void;

  // Offline indicator
  setShowOfflineIndicator: (show: boolean) => void;

  // Theme actions
  setTheme: (theme: 'light' | 'dark') => void;
  toggleTheme: () => void;
}

const useUIStore = create<UIStore>((set, get) => ({
  // Initial state
  selectedCategory: 'All',
  showPaymentModal: false,
  showOfflineIndicator: false,
  theme: 'light',

  // Category actions
  setSelectedCategory: (selectedCategory) => set({ selectedCategory }),

  // Modal actions
  setShowPaymentModal: (showPaymentModal) => set({ showPaymentModal }),

  // Offline indicator
  setShowOfflineIndicator: (showOfflineIndicator) => set({ showOfflineIndicator }),

  // Theme actions
  setTheme: (theme) => set({ theme }),
  toggleTheme: () =>
    set((state) => ({
      theme: state.theme === 'light' ? 'dark' : 'light',
    })),
}));

export default useUIStore;
