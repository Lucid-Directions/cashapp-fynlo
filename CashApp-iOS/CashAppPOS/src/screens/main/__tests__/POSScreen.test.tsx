/**
 * POSScreen Component Tests
 * Testing the main point-of-sale interface
 */

// @ts-nocheck
import React from 'react';
import { fireEvent, waitFor, act } from '@testing-library/react-native';
import POSScreen, { ExportedMenuItemCard } from '../POSScreen'; // Import ExportedMenuItemCard
import { customRenderWithStores, useTestTheme } from '../../../__tests__/utils/testUtils';
import useAppStore from '../../../store/useAppStore';
import useUIStore from '../../../store/useUIStore';
import { ThemeProvider } from '../../../design-system/ThemeProvider'; // Import ThemeProvider

// Mock navigation
const mockNavigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
  dispatch: jest.fn(),
};

// Mock menu items directly from POSScreen for consistency
const menuItems = [
  {
    id: 1,
    name: 'Nachos',
    price: 5.0,
    category: 'Snacks',
    emoji: '🧀',
    available: true,
    description: 'Description for Nachos',
  },
  {
    id: 6,
    name: 'Carnitas Taco',
    price: 3.5,
    category: 'Tacos',
    emoji: '🌮',
    available: true,
    description: 'Description for Carnitas',
  },
  {
    id: 7,
    name: 'Cochinita Taco',
    price: 3.5,
    category: 'Tacos',
    emoji: '🌮',
    available: true,
    description: 'Description for Cochinita',
  },
  {
    id: 20,
    name: 'Carne Asada Taco',
    price: 4.5,
    category: 'Special Tacos',
    emoji: '⭐',
    available: true,
    description: 'Description for Carne Asada',
  },
  {
    id: 31,
    name: 'Pink Paloma',
    price: 3.75,
    category: 'Drinks',
    emoji: '🍹',
    available: true,
    description: 'Description for Pink Paloma',
  },
  {
    id: 32,
    name: 'Coco-Nought',
    price: 3.75,
    category: 'Drinks',
    emoji: '🥥',
    available: false,
    description: 'Unavailable Drink',
  }, // Example of unavailable item
];

// Mock theme for ExportedMenuItemCard if needed, or wrap with ThemeProvider
const TestWrapper = ({ children }) => {
  const theme = useTestTheme(); // Assuming useTestTheme provides a valid theme object
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

describe('POSScreen', () => {
  // Initial store state that can be modified by tests
  let initialAppStoreState;
  let initialUIStoreState;

  beforeEach(() => {
    jest.clearAllMocks();

    // Reset store states before each test
    initialAppStoreState = {
      cart: [],
      addToCart: jest.fn(item =>
        useAppStore.setState(state => ({ cart: [...state.cart, { ...item, quantity: 1 }] })),
      ),
      removeFromCart: jest.fn(id =>
        useAppStore.setState(state => ({ cart: state.cart.filter(item => item.id !== id) })),
      ),
      updateCartItem: jest.fn((id, updates) =>
        useAppStore.setState(state => ({
          cart: state.cart.map(item => (item.id === id ? { ...item, ...updates } : item)),
        })),
      ),
      clearCart: jest.fn(() => useAppStore.setState({ cart: [] })),
      cartTotal: jest.fn(() =>
        useAppStore.getState().cart.reduce((sum, item) => sum + item.price * item.quantity, 0),
      ),
      cartItemCount: jest.fn(() =>
        useAppStore.getState().cart.reduce((sum, item) => sum + item.quantity, 0),
      ),
    };

    initialUIStoreState = {
      selectedCategory: 'All',
      setSelectedCategory: jest.fn(category => useUIStore.setState({ selectedCategory: category })),
      showPaymentModal: false,
      setShowPaymentModal: jest.fn(show => useUIStore.setState({ showPaymentModal: show })),
    };

    // Set initial state for stores
    useAppStore.setState(initialAppStoreState, true);
    useUIStore.setState(initialUIStoreState, true);
  });

  it('renders correctly and magnifier icon is not present', () => {
    const { getByText, getByTestId, queryByTestId, UNSAFE_getByProps } = customRenderWithStores(
      <POSScreen />,
      { navigationProps: { navigation: mockNavigation } },
    );

    // Check for some known text elements
    expect(getByText('Fynlo POS')).toBeTruthy(); // Assuming this is part of your header or a static text
    expect(getByTestId('menu-flat-list')).toBeTruthy();
    expect(UNSAFE_getByProps({ name: 'shopping-cart' })).toBeTruthy(); // Check for cart icon

    // Verify magnifier icon is not present
    // This depends on how the magnifier was implemented (e.g., by icon name or testID)
    // Assuming it was an Icon with name "search"
    const allIcons = UNSAFE_getByProps({ name: 'search' });
    // If the search icon was only in the header, and now we have one in the bubble,
    // we need a more specific way to check the header.
    // For now, let's assume the header search icon was unique or had a specific testID not present now.
    // If CategorySearchBubble also uses an icon named "search", this test needs adjustment.
    // Let's assume the old one was `header-search-icon` for clarity.
    expect(queryByTestId('header-search-icon')).toBeNull(); // Example testID for the old icon
  });

  it('displays menu items correctly', () => {
    const { getByText } = customRenderWithStores(<POSScreen />, {
      navigationProps: { navigation: mockNavigation },
    });
    expect(getByText('Nachos')).toBeTruthy();
    expect(getByText('£5.00')).toBeTruthy(); // Assuming formatPrice works like this
    expect(getByText('Carnitas Taco')).toBeTruthy();
  });

  it('adds item to cart when tapped', () => {
    const { getByText } = customRenderWithStores(<POSScreen />, {
      navigationProps: { navigation: mockNavigation },
    });

    const nachosItem = getByText('Nachos');
    fireEvent.press(nachosItem);

    expect(initialAppStoreState.addToCart).toHaveBeenCalledWith(
      expect.objectContaining({ id: 1, name: 'Nachos', price: 5.0 }),
    );
    // Check if cart state was updated (optional, as addToCart is mocked to update state)
    expect(useAppStore.getState().cart.length).toBe(1);
  });

  it('opens cart modal when cart icon is pressed', () => {
    const { getByTestId, getByText } = customRenderWithStores(<POSScreen />, {
      navigationProps: { navigation: mockNavigation },
    });
    const cartButton = getByTestId('shopping-cart-button');
    fireEvent.press(cartButton);
    expect(getByText('Current Order')).toBeTruthy(); // Modal title
  });

  it('CartIcon color is orange when empty, red when not empty', () => {
    const { getByTestId, UNSAFE_getByProps, rerender } = customRenderWithStores(<POSScreen />, {
      navigationProps: { navigation: mockNavigation },
    });

    // Initial state: cart is empty
    let cartIconComponent = UNSAFE_getByProps({ testID: 'shopping-cart-button' });
    let internalIcon = cartIconComponent.findAllByType(
      require('react-native-vector-icons/MaterialIcons').default,
    )[0];
    // Check for theme.colors.primary (Fynlo orange) - exact color depends on your theme setup
    // For this test, we'll assume the color prop is directly passed.
    // The theme object itself is complex to mock perfectly here, so we check the logic's output.
    // If theme.colors.primary is '#FF6B35' (example Fynlo Orange)
    expect(internalIcon.props.color).toBe('#FF6B35'); // Orange when empty

    // Add an item to the cart
    act(() => {
      useAppStore
        .getState()
        .addToCart({ id: 1, name: 'Test Item', price: 10, quantity: 1, emoji: '🧪' });
    });

    // Rerender or update component to reflect store changes
    rerender(
      <TestWrapper>
        <POSScreen navigation={mockNavigation} />
      </TestWrapper>,
    );

    cartIconComponent = UNSAFE_getByProps({ testID: 'shopping-cart-button' });
    internalIcon = cartIconComponent.findAllByType(
      require('react-native-vector-icons/MaterialIcons').default,
    )[0];
    expect(internalIcon.props.color).toBe('#FF3B30'); // Red when not empty
  });

  describe('Search Functionality', () => {
    it('renders CategorySearchBubble', () => {
      const { getByTestId } = customRenderWithStores(<POSScreen />, {
        navigationProps: { navigation: mockNavigation },
      });
      expect(getByTestId('category-search-bubble-inactive')).toBeTruthy();
    });

    it('expands search bubble on tap and shows TextInput', () => {
      const { getByTestId, getByPlaceholderText } = customRenderWithStores(<POSScreen />, {
        navigationProps: { navigation: mockNavigation },
      });
      const bubble = getByTestId('category-search-bubble-inactive');
      fireEvent.press(bubble);
      expect(getByTestId('category-search-bubble-active')).toBeTruthy();
      expect(getByPlaceholderText('Search food...')).toBeTruthy();
    });

    it('filters items when typing "Taco"', async () => {
      const { getByTestId, getByPlaceholderText, queryByText } = customRenderWithStores(
        <POSScreen />,
        { navigationProps: { navigation: mockNavigation } },
      );

      fireEvent.press(getByTestId('category-search-bubble-inactive'));
      const searchInput = getByPlaceholderText('Search food...');

      act(() => {
        fireEvent.changeText(searchInput, 'Taco');
      });

      // Wait for state update and re-render
      await waitFor(() => {
        expect(queryByText('Carnitas Taco')).toBeTruthy();
        expect(queryByText('Cochinita Taco')).toBeTruthy();
        expect(queryByText('Carne Asada Taco')).toBeTruthy();
        expect(queryByText('Nachos')).toBeNull(); // Should not be visible
        expect(queryByText('Pink Paloma')).toBeNull(); // Should not be visible
      });
    });

    it('clears search and shows all items (for "All" category) when clear button is pressed', async () => {
      const { getByTestId, getByPlaceholderText, queryByText } = customRenderWithStores(
        <POSScreen />,
        { navigationProps: { navigation: mockNavigation } },
      );

      fireEvent.press(getByTestId('category-search-bubble-inactive'));
      const searchInput = getByPlaceholderText('Search food...');

      act(() => {
        fireEvent.changeText(searchInput, 'Taco');
      });

      await waitFor(() => expect(queryByText('Nachos')).toBeNull()); // Pre-condition: Nachos is hidden

      const clearButton = getByTestId('clear-search-button');
      act(() => {
        fireEvent.press(clearButton);
      });

      await waitFor(() => {
        // Search query is cleared, should show all items if "All" category is selected
        expect(queryByText('Nachos')).toBeTruthy();
        expect(queryByText('Carnitas Taco')).toBeTruthy();
        expect(queryByText('Pink Paloma')).toBeTruthy();
      });
      // Bubble should also revert to inactive state
      expect(getByTestId('category-search-bubble-inactive')).toBeTruthy();
    });

    it('filters correctly when a category is selected and then search is used', async () => {
      const { getByTestId, getByPlaceholderText, getByText, queryByText } = customRenderWithStores(
        <POSScreen />,
        { navigationProps: { navigation: mockNavigation } },
      );

      // Select 'Tacos' category
      const tacosCategoryTab = getByText('Tacos');
      act(() => {
        fireEvent.press(tacosCategoryTab);
      });

      await waitFor(() => {
        expect(useUIStore.getState().selectedCategory).toBe('Tacos');
        expect(queryByText('Nachos')).toBeNull(); // Nachos (Snacks) should be hidden
        expect(queryByText('Carnitas Taco')).toBeTruthy();
        expect(queryByText('Carne Asada Taco')).toBeNull(); // Special Taco, should be hidden by category
      });

      // Now search within 'Tacos'
      fireEvent.press(getByTestId('category-search-bubble-inactive'));
      const searchInput = getByPlaceholderText('Search food...');
      act(() => {
        fireEvent.changeText(searchInput, 'Cochinita');
      });

      await waitFor(() => {
        expect(queryByText('Cochinita Taco')).toBeTruthy();
        expect(queryByText('Carnitas Taco')).toBeNull(); // Should be hidden by search
      });

      // Clear search, should revert to 'Tacos' category items
      const clearButton = getByTestId('clear-search-button');
      act(() => {
        fireEvent.press(clearButton);
      });

      await waitFor(() => {
        expect(queryByText('Carnitas Taco')).toBeTruthy();
        expect(queryByText('Cochinita Taco')).toBeTruthy();
        expect(queryByText('Carne Asada Taco')).toBeNull(); // Still hidden by category
        expect(queryByText('Nachos')).toBeNull(); // Still hidden by category
      });
    });
  });

  // Test for unavailable items (ensure it's not tappable)
  it('handles unavailable items correctly (not tappable)', () => {
    const { getByText } = customRenderWithStores(<POSScreen />, {
      navigationProps: { navigation: mockNavigation },
    });
    const unavailableItemText = getByText('Coco-Nought'); // This item is marked available: false

    // Check if the parent TouchableOpacity is disabled or press does not call addToCart
    // This depends on how ExportedMenuItemCard handles disabled state.
    // We'll assume if it's disabled, addToCart won't be called.
    fireEvent.press(unavailableItemText);
    expect(initialAppStoreState.addToCart).not.toHaveBeenCalledWith(
      expect.objectContaining({ name: 'Coco-Nought' }),
    );
  });
});

// Snapshot tests for Header Actions conditional rendering (if applicable)
// This part might need adjustment based on how IS_DEV and FLAGS are handled in your actual setup.
// For these tests, we'll mock them directly.
describe('POSScreen Header Actions Conditional Rendering Snapshots', () => {
  let mockIS_DEV;
  jest.mock('../../../env', () => ({
    // Corrected path
    get IS_DEV() {
      return mockIS_DEV;
    },
    envBool: jest.fn((name, fallback) => fallback),
  }));

  beforeEach(() => {
    jest.clearAllMocks();
    // @ts-ignore
    global.FLAGS = { SHOW_DEV_MENU: true }; // Default for these tests
  });

  it('snapshot: no bug icon in production (FLAGS.SHOW_DEV_MENU = true)', () => {
    mockIS_DEV = false;
    const { toJSON } = customRenderWithStores(<POSScreen />, {
      navigationProps: { navigation: mockNavigation },
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('snapshot: bug icon in development (FLAGS.SHOW_DEV_MENU = true)', () => {
    mockIS_DEV = true;
    const { toJSON } = customRenderWithStores(<POSScreen />, {
      navigationProps: { navigation: mockNavigation },
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('snapshot: no bug icon if FLAGS.SHOW_DEV_MENU = false (dev)', () => {
    mockIS_DEV = true;
    // @ts-ignore
    global.FLAGS.SHOW_DEV_MENU = false;
    const { toJSON } = customRenderWithStores(<POSScreen />, {
      navigationProps: { navigation: mockNavigation },
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('snapshot: no bug icon if FLAGS.SHOW_DEV_MENU = false (prod)', () => {
    mockIS_DEV = false;
    // @ts-ignore
    global.FLAGS.SHOW_DEV_MENU = false;
    const { toJSON } = customRenderWithStores(<POSScreen />, {
      navigationProps: { navigation: mockNavigation },
    });
    expect(toJSON()).toMatchSnapshot();
  });
});
