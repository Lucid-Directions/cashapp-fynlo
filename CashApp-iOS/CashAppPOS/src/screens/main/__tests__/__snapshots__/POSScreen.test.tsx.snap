// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POSScreen Header Actions Conditional Rendering Snapshots snapshot: bug icon in development (FLAGS.SHOW_DEV_MENU = true) 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`POSScreen Header Actions Conditional Rendering Snapshots snapshot: no bug icon if FLAGS.SHOW_DEV_MENU = false (dev) 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`POSScreen Header Actions Conditional Rendering Snapshots snapshot: no bug icon if FLAGS.SHOW_DEV_MENU = false (prod) 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`POSScreen Header Actions Conditional Rendering Snapshots snapshot: no bug icon in production (FLAGS.SHOW_DEV_MENU = true) 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`POSScreen Header Actions Conditional Rendering should not render bug icon if FLAGS.SHOW_DEV_MENU is false in production 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`POSScreen Header Actions Conditional Rendering should not render bug icon if FLAGS.SHOW_DEV_MENU is false, even in development 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`POSScreen Header Actions Conditional Rendering should not render bug icon in production when FLAGS.SHOW_DEV_MENU is true 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
